src/main/java/Controller/MainController.java

```diff
@@ -203,12 +207,18 @@ public void onDataChange(DataSnapshot snapshot) {
                                 databaseCommunicator.insertGroup(new Group(groupId, groupName,groupLink));
                             }
                             getGroups();
+                            done.countDown();
                         }
                         @Override
                         public void onCancelled(DatabaseError error) {
 
                         }
                     });
+                    try {
+                        done.await(); //it will wait till the response is received from firebase.
```

Comment:
Why do we need to block here?

Reply:
Due to the asynchronous nature of data retrieving in firebase. If we don't block here, then it can be possible that it shows sync completed even without fetching all the groups from the firebase database. In this situation, if we click the sync button again, then it leads to some miscellaneous behavior, it starts creating multiple instances of the same group. So, I think it's good to block here to stay out of any inconsistency.